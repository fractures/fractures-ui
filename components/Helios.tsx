import React from 'react'

const Helios = ({ colors }) => (
	<svg width="100" height="100" viewBox="0 0 100 100" fill="none" xmlns="http://www.w3.org/2000/svg">
		<path
			fillRule="evenodd"
			clipRule="evenodd"
			d="M15.4757 48.4914C15.0586 49.4141 15 49.897 15 50C15 50.103 15.0586 50.5859 15.4757 51.5086C15.8695 52.3795 16.493 53.4453 17.3632 54.6479C19.1013 57.0502 21.682 59.8037 24.9297 62.4069C31.4456 67.6298 40.3282 72 50 72C59.6718 72 68.5544 67.6298 75.0703 62.4069C78.318 59.8037 80.8987 57.0502 82.6368 54.6479C83.507 53.4453 84.1305 52.3795 84.5243 51.5086C84.9414 50.5859 85 50.103 85 50C85 49.897 84.9414 49.4141 84.5243 48.4914C84.1305 47.6205 83.507 46.5547 82.6368 45.3521C80.8987 42.9498 78.318 40.1963 75.0703 37.5931C68.5544 32.3702 59.6718 28 50 28C40.3282 28 31.4456 32.3702 24.9297 37.5931C21.682 40.1963 19.1013 42.9498 17.3632 45.3521C16.493 46.5547 15.8695 47.6205 15.4757 48.4914Z"
			fill={ colors.gray[100] }
		/>
		<path
			d="M34 50C34 41.1634 41.1634 34 50 34C58.8366 34 66 41.1634 66 50C66 58.8366 58.8366 66 50 66C41.1634 66 34 58.8366 34 50Z"
			fill={ colors.gray[500] }
		/>
		<circle cx="50" cy="50" r="9" fill={ colors.gray[900] } />
		<circle cx="58" cy="44" r="4" fill={ colors.gray[100] } />
		<circle cx="34" cy="50" r="3" fill={ colors.gray[100] } />
	</svg>
)

export default Helios
