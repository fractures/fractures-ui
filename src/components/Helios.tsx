import themes from '../config/themes'
import React from 'react'

const Helios = props => {
	const theme = themes.find(theme => props.theme === theme.name) || themes[0]

	// Colors
	const colors = Object.entries(theme.colors)
	const shades = colors.map((color, key) => {
		const colorName = colors[key][0]
		const colorShades = Object.entries(color[1])

		return colorShades.map(colorS => `--color-${ colorName }-${ colorS[0] }: ${ colorS[1] };`)
	})

	// Fonts
	const fonts = Object.entries(theme.fonts)
	const fontFamilies = fonts.map(font => {
		return `--font-${ font[0] }: ${ font[1] };`
	})

	const flatShades = shades.reduce((a, b) => a.concat(b)).reduce((a, b) => a.concat(b))
	const flatFontFamilies = fontFamilies.reduce((a, b) => a.concat(b))

	const themeCSS = `
		:root {
			${ flatShades }
			${ flatFontFamilies }
		}

		* {
			margin: 0;
			padding: 0;

			border: 0;
			box-sizing: border-box;
		}
	`

	return <style dangerouslySetInnerHTML={ { __html: themeCSS } } />
}

export default Helios
