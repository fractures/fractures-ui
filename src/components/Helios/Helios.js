import themes from "../../config/themes"
import PropTypes from "prop-types"
import React from "react"

const propTypes = {
	theme: PropTypes.oneOf(["blue", "black", "invert", "test"])
}

const defaultProps = {
	theme: null
}

// NOTE This is still WIP
// TODO Enable custom themes, finalize structure.

// Parsing themes to be able to inject them to :root.
const Helios = props => {
	const theme = themes.find(theme => props.theme === theme.name)

	if(!theme) return null

	// Colors
	const colors = Object.entries(theme.colors)
	const shades = colors.map((color, key) => {
		const colorName = colors[key][0]
		const colorShades = Object.entries(color[1])

		return colorShades.map(colorS => `--color-${ colorName }-${ colorS[0] }: ${ colorS[1] };`)
	})

	// Fonts
	const fonts = Object.entries(theme.fonts)
	const fontFamilies = fonts.map(font => {
		return `--font-${ font[0] }: ${ font[1] };`
	})

	const themeCSS = `
		:root {
			${ shades.flat(2).join("\n") }
			${ fontFamilies.flat().join("\n") }
		}
	`

	return <style dangerouslySetInnerHTML={ { __html: themeCSS } } />
}

Helios.propTypes = propTypes
Helios.defaultProps = defaultProps

export default Helios
