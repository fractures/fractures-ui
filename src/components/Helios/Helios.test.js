import Helios from "./Helios.js"

describe("Helios", () => {
	const tree = renderer.create(<Helios />).toJSON()

	it("Renders correctly", () => expect(tree).toMatchSnapshot())
})

describe("Helios with theme", () => {
	const tree = renderer.create(<Helios theme="blue" />).toJSON()

	it("Renders correctly", () => expect(tree).toMatchSnapshot())
})
