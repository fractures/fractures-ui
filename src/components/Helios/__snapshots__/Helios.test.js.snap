// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Helios Renders correctly 1`] = `null`;

exports[`Helios with theme Renders correctly 1`] = `
<style
  dangerouslySetInnerHTML={
    Object {
      "__html": "
		:root {
			--color-gray-0: #fff;--color-gray-100: #e8f2fd;--color-gray-300: #76a7e7;--color-gray-500: #3482da;--color-gray-700: #2463a5;--color-gray-900: #102a45;--color-red-100: #FFD8DA;--color-red-300: #FF979D;--color-red-500: #f94b56;--color-red-700: #CF2E38;--color-red-900: #8F171F;--color-yellow-100: #FFF2D2;--color-yellow-300: #FFE6A6;--color-yellow-500: #FFD058;--color-yellow-700: #DFA103;--color-yellow-900: #8D5D00;--color-green-100: #E4FBDF;--color-green-300: #B3E8A7;--color-green-500: #82C971;--color-green-700: #4DA439;--color-green-900: #1A6309;--color-blue-100: #D3E6FF;--color-blue-300: #95C3FF;--color-blue-500: #539DFE;--color-blue-700: #216FD3;--color-blue-900: #073775;
			--font-default: Roboto, 'sans-serif';--font-mono: 'Roboto Mono', monospace;
		}
	",
    }
  }
/>
`;
